src/lfortran/semantics/python_ast_to_asr.cpp

```diff
@@ -1212,6 +1212,29 @@ class BodyVisitor : public CommonVisitor<BodyVisitor> {
             } else {
                 throw SemanticError("chr() must have one integer argument", x.base.base.loc);
             }
+        } else if (call_name == "complex") {
+            LFORTRAN_ASSERT(ASRUtils::all_args_evaluated(args));
```

Comment:
I think that if variables are allowed then this check should be removed as this accepts only those arguments which are constants or can be evaluated to a constant at compile time.

Reply:
Yup, I agree with you! 
But, I think naman is planning to support only constant values for now. (I don't know)

Reply:
In that case, we can add a comment saying "Only constant numeric values as parameter are supported for now". Iteratively we can support for both strings(`complex('1+2j')` or `complex('1')`) and variables (`complex(x)`)

Reply:
Thanks.  

> I think that if variables are allowed then this check should be removed as this accepts only those arguments which are constants or can be evaluated to a constant at compile time.

Yes, this should be done. I guess we also need to modify other intrinsic functions in this file (Python AST->ASR), like `chr` and `ord`, in order to make support for variables.